crates/cli/src/commands/prove.rs

```diff
@@ -103,11 +114,34 @@ enum ProveSubCommand {
         #[command(flatten)]
         cargo_args: RunCargoArgs,
 
+        #[command(flatten)]
+        segmentation_args: SegmentationArgs,
+
         #[command(flatten)]
         agg_tree_config: AggregationTreeConfig,
     },
 }
 
+#[derive(Clone, Copy, Parser)]
+pub struct SegmentationArgs {
+    /// Trace height threshold, in bits, across all chips for triggering segmentation for
+    /// continuations in the app proof. Note that these thresholds are not absolute limits.
+    #[arg(
+        long,
+        default_value_t = DEFAULT_MAX_TRACE_HEIGHT_BITS,
+        help_heading = "OpenVM Options"
+    )]
+    pub segment_max_height_bits: u8,
```

Comment:
can you also add a `debug_assert` somewhere in metered ctx that the `max_trace_height` value used is a power of 2

Reply:
https://github.com/openvm-org/openvm/pull/2178/commits/27691371341518c7a2c783b5875b5c6dc47cd4d4